pkg/cmd/miscellaneous.go

```diff
@@ -350,17 +361,84 @@ func getRole() string {
 }
 
 /*
-getTechnicalID returns the Technical Id, which used as clustername of the shoot cluster
+getTargetMapInfo retun garden,project,seed,shoot,TechnicalID to global getTargetInfo
 */
-func getTechnicalID() string {
+func getTargetMapInfo() map[string]string {
 	var target Target
 	ReadTarget(pathTarget, &target)
 
-	gardenClientset, err := target.GardenerClient()
-	checkError(err)
-	project, err := gardenClientset.CoreV1beta1().Projects().Get(target.Stack()[1].Name, metav1.GetOptions{})
-	checkError(err)
-	shoot, err := gardenClientset.CoreV1beta1().Shoots(*project.Spec.Namespace).Get(target.Stack()[2].Name, metav1.GetOptions{})
-	checkError(err)
-	return shoot.Status.TechnicalID
+	for _, t := range target.Stack() {
+		getTargetInfo[string(t.Kind)] = string(t.Name)
+	}
+
+	// user for case garden-[project]-seed-shoot
+	if isTargetFilsHas("seed", "shoot") {
+		delete(getTargetInfo, "seed")
+
+		shootname := getTargetInfo["shoot"]
+		project, err := getProjectObject()
+		checkError(err)
+		shoot, err := getShootObject(*project.Spec.Namespace, shootname)
+		checkError(err)
+		getTargetInfo["seed"] = *shoot.Spec.SeedName
+		getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+		shoot, err = getShootObject("garden", getTargetInfo["seed"])
+		checkError(err)
+		getTargetInfo["seedTechnicalID"] = shoot.Status.TechnicalID
```

Comment:
> You can't always assume that you find a shoot resource for a seed.
> Also `seedTechnicalID` is currently not used anyhow. Can't the logic to determine `seedTechnicalID` be removed?

fixed 

> In addition, getting the shoot in the garden namespace would fail for an end-user/non-operator. 

I am not following this part.  The logic from my side is,  

from line 379 
`getProjectObject` will return ProjectObject either operator or end-user role. which include `*Spec.Namespace` eg `garden-i333878' for my case as non-operator role.

getShootObject will return ShootObject base on garden kubeconfig level `gardenClientset, err := target.GardenerClient()` 

```
		project, err := getProjectObject()
		checkError(err)
		shoot, err := getShootObject(*project.Spec.Namespace, shootname)
		checkError(err)
```

![image](https://user-images.githubusercontent.com/42234376/93837668-a411ca00-fcb9-11ea-920d-93dc8079142d.png)